src/HQDM.ts

```diff
@@ -83,6 +104,70 @@ export class HQDMModel {
   constructor() {
     this.things = new Map();
     this.relations = new Map();
+    this.setOurVersionInfo();
+  }
+
+  /**
+   * Ensure output files include a given owl:versionInfo triple. This
+   * will remove any existing version triple for this subject.
+   *
+   * @param subject The subject IRI of the triple.
+   * @param version The version number to write as the object.
+   */
+  setVersionInfo(subject: string, version: string) {
+    const s_th = new Thing(subject);
+    const v_th = new Thing(version);
+    this._getRelated(s_th, OWL_VERSION)
+      .forEach((v: Thing) => this.unrelate(OWL_VERSION, s_th, v));
+    this.relate(OWL_VERSION, s_th, v_th);
+  }
+
+  /**
+   * Check if we have a given owl:versionInfo triple. This may have been
+   * created with setVersionInfo or loaded from a file.
+   *
+   * @param subject The subject IRI to look for.
+   * @returns The version string found, or undefined.
+   */
+  getVersionInfo(subject: string) {
+    return this._getRelated(new Thing(subject), OWL_VERSION)
+      .first()?.id;
+  }
+
+  /**
+   * Set the hqdm-lib version info. Currently this sets the HQDM RDF
+   * representation version for forward-compat and the version of
+   * hqdm-lib just for debugging/interest.
+   */
+  setOurVersionInfo() {
+    this.setVersionInfo(HQDM_LIB_REPR, currentReprVersion);
+    this.setVersionInfo(HQDM_LIB_NS, HQDM_LIB_VERSION);
+  }
+
+  replaceIriPrefix(from: string, to: string) {
```

Comment:
The client in this case is ActivityLib; I would say 'would you prefer this functionality go into ActivityLib' but that isn't possible because it requires access to hqdm-lib internals. I am quite sure we will need this sort of transformation again in situations where more people than just us have TTL files with valuable work in, and expecting end-users to manipulate TTL files by hand is IMHO unreasonable.

I will leave it in, but I will note in the documentation that it may go away in the future.

I think the root problem here is that hqdm-lib is doing two jobs: it's a triplestore, and it's an HQDM datastore built on top of that. The versioning stuff and this replace-prefix method are reasonable utility methods on a triplestore, though not so much on an HQDM datastore.

In an ideal world, I would like clients (ActivityLib) to be able to
* Load turtle/json-ld into the triplestore.
* Ask questions of the triplestore, e.g. about version relations.
* If necessary perform data fixups on the triples at this stage.
* Then generate the HQDM model from the triplestore.

But we're not going to achieve that in three days.

Reply:
There is a Shortcut story about splitting the triplestore part out to a separate library, so I agree with you there.

As far as I know if you wanted to do something similar with Jena you'd have to use SPARQL to extract, transform, and load the data, so ActivityLib could convert a model to a TTL string, do a string replace, and convert back to a model. I think my concern is that converting old data to new data isn't the concern of this library.

I'll leave it up to you though.